magefiles/utils.go

```diff
@@ -230,3 +231,56 @@ func cleanupQuay(quayService quay.QuayService, quayOrg string) error {
 	}
 	return nil
 }
+
+func cleanupQuayTags(quayService quay.QuayService, organization, repository string) error {
+	workerCount := 10
+	tagsChan := make(chan []quay.Tag, workerCount)
+	var wg sync.WaitGroup
+
+	var errors []error
+
+	for i := 0; i < workerCount; i++ {
+		wg.Add(1)
+		go func(tagsChan <-chan []quay.Tag, wg *sync.WaitGroup) {
+			defer wg.Done()
+
+			for tags := range tagsChan {
+				for _, tag := range tags {
+					if time.Unix(tag.StartTS, 0).Before(time.Now().AddDate(0, 0, -7)) {
+						deleted, err := quayService.DeleteTag(organization, repository, tag.Name)
+						if err != nil {
+							errors = append(errors, err)
+						}
+						if !deleted {
+							fmt.Printf("tag `%s` was not deleted\n", tag.Name)
+						}
+					}
+				}
+			}
+		}(tagsChan, &wg)
+	}
+
+	page := 1
+	for {
+		tags, hasAdditional, err := quayService.GetTagsFromPage(organization, repository, page)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		tagsChan <- tags
+		if !hasAdditional {
+			break
+		}
+		page++
+	}
+	close(tagsChan)
+	wg.Wait()
+
+	if len(errors) == 0 {
+		return nil
+	}
+	for _, err := range errors {
+		fmt.Printf("encountered error during CleanupQuayTags: %v\n", err)
+	}
+	return fmt.Errorf("encountered errors above")
```

Comment:
you can use methods from errors package and do something like
```go
// rename `errors []error` e.g. to  `errSlice []error` and then
return fmt.Errorf("encountered errors during CleanupQuayTags: %v", errors.Join(errSlice...))
```

Reply:
The project is [using Go 1.19](https://github.com/redhat-appstudio/e2e-tests/blob/33ca54b2527b8a75b236c0c14186f6d847be9f5e/go.mod#L3), isn't it?
`errors.Join()` was [introduced in Go 1.20](https://pkg.go.dev/errors#Join)

Reply:
haha, that's a good point, didn't notice that it was introduced so late.

other option is to collect the slice of strings [here](https://github.com/redhat-appstudio/e2e-tests/pull/467/files#diff-8d3f02a956e3c919ac5ea21a43aa554db0d763ffa49bac99f9f294d88e612d59R267) and then use `strings.Join`... or it doesn't actually have to be a slice, it could be a just a string, that would be just printed here...